Assets/Scripts/Scene Controller.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+[CreateAssetMenu(fileName = "SceneController", menuName = "Scriptable Objects/SceneController")]
+public class SceneController : ScriptableObject
+{
+    public static SceneController instance;
+
+    [SerializeField] private string calendar;
+    [SerializeField] private string awake;
+
+
+
+    private List<string> scenesToUnload;
+
+    public void Initialise()
+    {
+        instance = this;
+
+        scenesToUnload = new List<string>();
+        EventManager.instance.onOpenGame += HandleOpenGame;
+
+    }
+
+
+
+
+
+
+
+    // --------------------------------------------------------------------------------------
+    // Scene Controller-speciefic functions
+    // --------------------------------------------------------------------------------------
+
+    public void HandleOpenGame()
+    {
+
+        LoadScene(calendar, true);
+        scenesToUnload.Add(awake);
+        UnloadScenes();
+    }
+
+
+
+
+    public void LoadScene(string name, bool isPermanent)
+    {
+        SceneManager.LoadSceneAsync(name, LoadSceneMode.Additive);
+        if (!isPermanent)
+        {
+            scenesToUnload.Add(name);
+        }
+        Debug.Log("pop" + Time.frameCount);
```

Comment:
Unneeded Debug.Log - when you write to console its an expensive operation. Fairly negligible here, but its worth taking out these logs if they aren't necessary.

Reply:
Resolved